spec/api-subframe-spec.js

```diff
@@ -0,0 +1,88 @@
+const { expect } = require('chai')
+const { remote } = require('electron')
+const path = require('path')
+
+const { emittedNTimes, emittedOnce } = require('./events-helpers')
+const { closeWindow } = require('./window-helpers')
+
+const { BrowserWindow } = remote
+
+describe('renderer nodeSupportInSubFrames', () => {
+  const generateTests = (sandboxEnabled) => {
+    describe(`with sandbox ${sandboxEnabled ? 'enabled' : 'disabled'}`, () => {
+      let w
```

Comment:
instead of this shared global, perhaps we could do something like

```js
const withWindow = async (f) => {
  const w = new BrowserWindow(...)
  await f(w)
  await closeWindow(w)
}

// later

it('something', async () => {
  await withWindow(async (w) => {
    // ...
  })
})
```

Or perhaps there's a nice way to do something like 

```js
it('something', withWindow(async () => {
  // ...
}))
```

mainly I don't like that the window is closed in the test _after_. That could result in test failures relating to teardown being misattributed.

Reply:
This is our current pattern, I like the idea of `withWindow` but I'd rather do that as a sweeping change rather than file-by-file for consitency 👍